### ISSUE TITLE
XSS Vulnerability in Markdown Editor

### ISSUE THREAD

**Issue by llamakko** [hackmdio/hackmd#412]
I found a XSS vulnerability in the following place.

https://github.com/hackmdio/hackmd/blob/master/public/js/render.js#L7

```
var linkRegex = /^(?!javascript:\/\/)([\w|-]+:\/\/)|^([.|/])+/
```

Steps to reproduce:

1. Add the following code to Markdown Editor.

```
<iframe src=Javascript://%0aalert(document.domain)></iframe>
```

2. Will display an alert dialog containing the document.domain.

---

I think that this place had better use a whitelist of URI protocols (e.g., http, https) and relative path unless there is some particular reason.

**Comment by attritionorg** [comment]
Dupe to https://github.com/hackmdio/hackmd/issues/396?

**Comment by SISheogorath** [comment]
The fix is incomplete. So yes it's more or less a duplicate. The original bug should be reopened and fixed by at least adding an `i` at the end, to make it case insensitive.

I'll file a pull request later if it's not already fixed then.

**Comment by jackycute** [comment]
Hi @llamakko 
Thanks for reporting this.
I didn't think of the capital `"J"avscript` is also working.
So, yeah @SISheogorath case insensitive flag would definitely help.

The reason why not only whitelist `http` and `https` because there are more protocols like `mms`, `rtsp`.
And even some custom protocol to activate external application.
In this situation, we would rather use blacklist to ban vulnerable protocols.

**Comment by jackycute** [comment]
Fixed in 48df2504914d638d974e14c07908230d4e971211
